src/main/resources/dbreplica/migration/V1.0.10__configure_replication_plsql.sql

```diff
@@ -0,0 +1,183 @@
+          update_action = l_task.last_updated_action,
+          due_date = l_task.due_date_time::date,
+          last_updated_date = l_task.last_updated_timestamp::date,
+          completed_date = case when (l_task.last_updated_action='Complete') and (reportable_task.completed_date is null) then l_task.last_updated_timestamp::date end,
+          completed_date_time = case when (l_task.last_updated_action='Complete') and (reportable_task.completed_date is null) then l_task.last_updated_timestamp end,
+          final_state_label =
+              case
+                when (l_task.last_updated_action='Complete') and (reportable_task.final_state_label is null) then 'COMPLETED'
+                when (l_task.last_updated_action='Cancel') and (reportable_task.final_state_label is null) then 'USER_CANCELLED'
+                when (l_task.last_updated_action='AutoCancel') and (reportable_task.final_state_label is null) then 'AUTO_CANCELLED'
+              end,
+          first_assigned_date =  case when (reportable_task.first_assigned_date is null) and (l_task.assignee is not null) then l_task.last_updated_timestamp::date else reportable_task.first_assigned_date end,
+          first_assigned_date_time = case when (reportable_task.first_assigned_date_time is null) and (l_task.assignee is not null) then l_task.last_updated_timestamp else reportable_task.first_assigned_date_time end,
+          wait_time_days = case when (l_task.last_updated_action='Complete') then (l_task.last_updated_timestamp::date - reportable_task.created_date) end,
+          handling_time_days = case when (l_task.last_updated_action='Complete') then (l_task.last_updated_timestamp::date - l_task.last_updated_timestamp::date) end,
+          processing_time_days = case when (l_task.last_updated_action='Complete') then (l_task.last_updated_timestamp::date - l_task.created::date) end,
+          is_within_sla = case when (l_task.last_updated_action='Complete') and (l_task.last_updated_timestamp <= l_task.due_date_time) then 'Yes'
+                                when (l_task.last_updated_action='Complete') and (l_task.last_updated_timestamp > l_task.due_date_time) then 'No' end,
+          number_of_reassignments = (select count(*) -1 from cft_task_db.task_history where task_history.state = 'ASSIGNED' and task_history.task_id =  l_task.task_id),
+          due_date_to_completed_diff_days = case when (l_task.last_updated_action='Complete') then (l_task.last_updated_timestamp::date - l_task.due_date_time::date) end
+    where reportable_task.task_id = l_task.task_id;
+  end if;
+
+return l_update_id;
+end $function$;
+
+--
+-- Function to call from triggers whenever a task record is inserted or updated.
+--
+create or replace function on_task_insert()
+  returns trigger
+  language plpgsql
+as $function$
+begin
+  perform cft_task_db.add_task_history(new, true);
+return new;
+end $function$;
+
+create or replace function on_task_update()
```

Comment:
Is this one used?

Reply:
can't see anywhere where this is called...

Reply:
As discussed with Ravi, this will be refactored RWA-2085

Reply:
removed